doc/nrf/protocols/bt/ble/conn_evt_cb.rst

```diff
@@ -0,0 +1,37 @@
+.. _ug_bt_conn_evt_cb:
+
+Synchronize data sampling with Bluetooth connection events
+##########################################################
+
+The Bluetooth connection event callback feature is used to inform the application about connection events.
+The connection event prepare callback triggered right before of a Bluetooth connection event.
+This allows the application to trigger sensor data collection for transmission during the upcoming connection event.
+
+.. figure:: images/ConnEvtCb.svg
+   :alt: Connection event prepare callback
+
+   Connection event prepare callback is called before a connection event
+
+The timing of the prepare callback is configured with the parameter `prepare_time_us`.
+The parameter must be set sufficiently large to allow the Bluetooth Stack to enqueue a data packet for the upcoming connection event.
+
+The :ref:`ble_conn_evt_cb` sample application demonstrates how this feature is used to reduce the latency between data sampling and data transmission.
+
+Connection event callback with peripheral latency
+*************************************************
+
+When a connection is configured with peripheral latency, the peripheral device may sleep for `peripheral_latency` consecutive connection events to save power.
+It may wake up more often if, for example if it has data to send.
+Therefore, the central device assumes the peripheral will wake up every connection event.
+
+The connection event prepare callback will be called for every connection event for both the peripheral and central device.
+If the peripheral application provides data to be sent in this callback, the peripheral device will wake up for that given connection event.
+
+.. figure:: images/ConnEvtCbPeripheralLatency.svg
+   :alt: Connection event prepare callback with peripheral latency
+
+   The peripheral wakes up if data is provided in a connection event prepare callback
+
+For this configuration, the application is required to configure the prepare callback to trigger at least 3 ms before the connection event starts.
```

Comment:
Should we add a constant somewhere for these 3ms?
Like that, we could change it more easily, if needed (e.g. to account for different HWs)

Reply:
Yes, I believe we can add something like `BT_CONN_EVT_CB_PREPARE_DISTANCE_US_RECOMMENDED` or something similar to `evt_cb.h`